filament/src/MaterialInstance.cpp

```diff
@@ -177,24 +177,39 @@ template UTILS_PUBLIC void MaterialInstance::setParameter<mat4f>   (const char*
 
 FMaterialInstance::FMaterialInstance() noexcept = default;
 
-FMaterialInstance::FMaterialInstance(FEngine& engine, FMaterial const* material, const char* name) :
-        mName(name) {
+FMaterialInstance::FMaterialInstance(FEngine& engine,
+        FMaterialInstance const* other, const char* name)
+        : mPolygonOffset(other->mPolygonOffset),
+          mCulling(other->mCulling),
+          mColorWrite(other->mColorWrite),
+          mDepthWrite(other->mDepthWrite),
+          mDepthFunc(other->mDepthFunc),
+          mScissorRect(other->mScissorRect),
+          mName(name ? CString(name) : other->mName) {
+
     FEngine::DriverApi& driver = engine.getDriverApi();
+    FMaterial const* const material = other->getMaterial();
 
     if (!material->getUniformInterfaceBlock().isEmpty()) {
-        mUniforms.setUniforms(material->getDefaultInstance()->getUniformBuffer());
+        mUniforms.setUniforms(other->getUniformBuffer());
         mUbHandle = driver.createUniformBuffer(mUniforms.getSize(), backend::BufferUsage::DYNAMIC);
     }
 
     if (!material->getSamplerInterfaceBlock().isEmpty()) {
-        mSamplers.setSamplers(material->getDefaultInstance()->getSamplerGroup());
+        mSamplers.setSamplers(other->getSamplerGroup());
         mSbHandle = driver.createSamplerGroup(mSamplers.getSize());
     }
 
     initialize(material);
```

Comment:
Inside `initialize` there are calls like `setMaskThreshold(material->getMaskThreshold());` that to my understanding modify the uniforms. If we do something like this:
```C++
MaterialInstance *mi = material->createInstance("original");
mi->setMaskThreshold(0.5f);
MaterialInstance duplicate = mi->duplicate("duplicate");
```
wouldn't we expect the mask threshold of the duplicate to be the overridden value instead of the material's default?

Reply:
yes very good point, I suppose we don't want the value from the material in that case.